src/massive/munit/Assert.hx

```diff
@@ -289,29 +296,134 @@ class Assert
 		return null;
 	}
 
+	/**
+	 * Assert that a string or iterable is empty
+	 * @param anObject The string or iterable to be tested
+	 * @param message The message to display in case of failure
+	 * @throws AssertionException if value is not empty
+	 */
+	public static function isEmpty(anObject:StringOrIterable, ?message:String, ?info:PosInfos) {
+		if(empty(anObject)) return;
+		if(message == null) message = 'Value [${anObject}] was not EMPTY';
+		fail(message, info);
+	}
+	
+	/**
+	 * Assert that a string or iterable is not empty
+	 * @param anObject The string or iterable to be tested
+	 * @param message The message to display in case of failure
+	 * @throws AssertionException if value is empty
+	 */
+	public static function isNotEmpty(anObject:StringOrIterable, ?message:String, ?info:PosInfos) {
+		if(!empty(anObject)) return;
+		if(message == null) message = 'Value [${anObject}] was EMPTY';
+		fail(message, info);
+	}
+	
 	/**
 	 * Force an assertion failure.
 	 * 
 	 * @param	message				message describing the assertion which failed
 	 * @throws	AssertionException	thrown automatically
 	 */
-	public static function fail(message:String, ?info:PosInfos)
-	{
-		throw new AssertionException(message, info);
-	}
+	public static function fail(message:String, ?info:PosInfos) throw new AssertionException(message, info);
 	
-	static inline function equals(a:Dynamic, b:Dynamic) return switch(Type.typeof(a)) {
-		case TEnum(_): Type.enumEq(a, b);
-		case TFunction: Reflect.compareMethods(a, b);
-		default:
-			#if(js)
-				#if(haxe_ver >= "4.0.0")
-				js.Syntax.strictEq(a, b);
+	static function equals(a:Dynamic, b:Dynamic):Bool {
+		return switch(Type.typeof(a)) {
+			case TEnum(_): Type.enumEq(a, b);
+			case TFunction: Reflect.compareMethods(a, b);
+			case TClass(_):
+				if(Std.is(a, String) && Std.is(b, String)) return a == b;
+				if(Std.is(a, Array) && Std.is(b, Array)) {
+					var a:Array<Dynamic> = cast a;
+					var b:Array<Dynamic> = cast b;
+					if(a.length != b.length) return false;
+					for(i in 0...a.length) {
+						if(!equals(a[i], b[i])) return false;
```

Comment:
I think I merged too quickly: I think it should be doing `areSame` comparisons and not a full deep comparison.

Reply:
for this key: `[[1]]` or this `[{x:1}]` or: a:[1] , b:["1"]

Reply:
IMHO content comparison should be shallow - when comparing values of the array or fields it should be strict equality.

Reply:
The current implementation will definitely not make it worse;)